### ISSUE TITLE
Annotation in extended class

### ISSUE THREAD

**Issue by theseeker58** [fillumina/krasa-jaxb-tools#16]
Hello there

I'm trying to generate jaxb classes from a xsd schema (which I don't want to tamper) using hisrc-higherjaxb40-maven-plugin from Patrodyne and krasa-jaxb-tools for automatically generating bean validation annotations from schema restrictions.
Everything works fine but the following extended class from an extended complex type.

```
  <xs:complexType name="ActiveOrHistoricCurrencyAndAmount">
    <xs:simpleContent>
      <xs:extension base="ActiveOrHistoricCurrencyAndAmount_SimpleType">
        <xs:attribute name="Ccy" type="ActiveOrHistoricCurrencyCode" use="required"/>
      </xs:extension>
    </xs:simpleContent>
  </xs:complexType>

  <xs:complexType name="ActiveOrHistoricCurrencyAndAmount_EPC258-22_V3.0_DS01_2">
    <xs:simpleContent>
      <xs:restriction base="ActiveOrHistoricCurrencyAndAmount">
        <xs:maxInclusive value="999999999.99"/>
        <xs:minInclusive value="0.01"/>
        <xs:fractionDigits value="2"/>
        <xs:attribute name="Ccy" type="ActiveOrHistoricCurrencyCode" use="required"/>
      </xs:restriction>
    </xs:simpleContent>
  </xs:complexType>

  <xs:simpleType name="ActiveOrHistoricCurrencyAndAmount_SimpleType">
    <xs:restriction base="xs:decimal">
      <xs:minInclusive value="0"/>
      <xs:totalDigits value="18"/>
      <xs:fractionDigits value="5"/>
    </xs:restriction>
  </xs:simpleType>

```
xjc generates the class ActiveOrHistoricCurrencyAndAmount where the simple value is correctly annotated
```
    @XmlValue
    @DecimalMin(value = "0", inclusive = true)
    @Digits(integer = 18, fraction = 5)
    protected BigDecimal value;
```

In the derived class ActiveOrHistoricCurrencyAndAmountEPC25822V30DS012 the value property is not overridden using the different restriction from the xsd schema. Am I missing anything in configuration? Or is it a bug?

```
@XmlAccessorType(XmlAccessType.FIELD)
@XmlType(name = "ActiveOrHistoricCurrencyAndAmount_EPC258-22_V3.0_DS01_2")
public class ActiveOrHistoricCurrencyAndAmountEPC25822V30DS012
    extends ActiveOrHistoricCurrencyAndAmount
    implements Serializable
{

    private static final long serialVersionUID = 1L;

}
```
Many thanks

**Comment by fillumina** [comment]
Hi @theseeker58,
unfortunately this issue relates to a XJC difficulty to handle that particular kind of extension. It is in fact tricky and difficult to resolve in java code. The match between xsd and java is not perfect and there are some corner cases which can not (or are not) handled by XJC.

I have found this stack overflow article that can better explain your situation: https://stackoverflow.com/questions/26618647/jaxb2-type-restriction-not-working

Being the issue not related to krasa-jaxb-tools I am closing it, feel free to reopen it in case my assumptions were wrong or I had misinterpreted the problem.

Thank you

**Comment by fillumina** [comment]
I have add your example to a test in commit 632bd106c3fc6e566a6ba153906936dc7bf4ec78

**Comment by theseeker58** [comment]
Many thanks @fillumina for your quick response.
This is how I solved the issue. It seems to me the less dirty way.
Considering that luckily just one (out of 140) xml type is affected, I have written the mapping class by hand, literally copying the one generated by xjc, but annotating the value field with the stricter rules.

```
    @XmlValue
    @DecimalMin(value = "0.01", inclusive = true)
    @Digits(integer = 18, fraction = 2)
    protected BigDecimal value;

```

Finally, in the xjb customization file I have used the ref attribute to indicate the already existing mapping class.

```
        <jaxb:bindings node="xsd:complexType[@name='ActiveOrHistoricCurrencyAndAmount']">
          <jaxb:class ref="com.fsajeva.rtp.iso.v10.payee.pain013.ActiveOrHistoricCurrencyAndAmount"/>
        </jaxb:bindings>
```

Honestly I can't see a better solution than this. The alternative is the xml schema validation. Quite easy to perform, but the drawback is the errors expressiveness that is definitely poorer if compared to the bean validation.
Thanks again
